### PR TITLE
resolved a lint error and handled some ignored errors in file_indexer

### PR DESCRIPTION
**Pull request by girishramnani**
## What is the purpose of this change? What does it change?
resolved a lint error and handled some ignored errors in file_indexer

## Was the change discussed in an issue?
No issue

## How to test changes?
just check if the index file is created correctly

### CODE REVIEW

**Comment by amitkrout** [comment]
> just check if the index file is created correctly

@girishramnani Which index file and how ?

**Comment by dharmit** [comment]
> > just check if the index file is created correctly
> 
> @girishramnani Which index file and how ?

+1. Have the same query

File: pkg/util/file_indexer.go [hunk 1]
```diff
@@ -149,17 +146,9 @@ func Run(directory string, ignoreRules []string) (filesChanged []string, filesDe
 // filepath is the location of the file to which it is supposed to be written
 func write(filePath string, writeMap map[string]FileData) error {
 	jsonData, err := json.Marshal(writeMap)
```
Comment (girishramnani) [inline pkg/util/file_indexer.go]: two `err` one after the other, but error checked in the end. which means this error will be ignore

File: pkg/util/file_indexer.go [hunk 2]
```diff
@@ -126,7 +123,7 @@ func Run(directory string, ignoreRules []string) (filesChanged []string, filesDe
 	}
 
 	// find files which are deleted/renamed
-	for fileName, _ := range readData {
```
Comment (girishramnani) [inline pkg/util/file_indexer.go]: lint: second arg doesn't need to be ignored. we can just eliminate it

File: pkg/util/file_indexer.go [hunk 3]
```diff
@@ -26,14 +27,10 @@ func read(filePath string) (map[string]FileData, error) {
...
 
-	jsonFileRead, err := os.Open(filePath)
-	// if we os.Open returns an error then handle it
+	byteValue, err := ioutil.ReadFile(filePath)
 	if err != nil {
 		return nil, err
 	}
-	defer jsonFileRead.Close()
-
-	byteValue, _ := ioutil.ReadAll(jsonFileRead)
```
Comment (girishramnani) [inline pkg/util/file_indexer.go]: fix: ignored the error here

**Comment by girishramnani** [comment]
@amitkrout @dharmit `.odo/odo-file-index.json`, introducted as part of the smart odo push https://github.com/openshift/odo/pull/1849

**Comment by girishramnani** [comment]
@amitkrout I think it would be good idea to add that as part of a separate PR. I can record this in an issue. But if you feel we need it as part of this PR then I can surely add it.

**Comment by amitkrout** [comment]
> I think it would be good idea to add that as part of a separate PR.

Works for me. Please create a separate issue and assign to me. I will take care of that

/approve

File: pkg/util/file_indexer.go [hunk 4]
```diff
@@ -26,14 +27,10 @@ func read(filePath string) (map[string]FileData, error) {
 		return map[string]FileData{}, nil
 	}
 
-	jsonFileRead, err := os.Open(filePath)
-	// if we os.Open returns an error then handle it
+	byteValue, err := ioutil.ReadFile(filePath)
```
Comment (mik-dass) [inline pkg/util/file_indexer.go]: please add some comments explaining the benefit of this approach as compared to the earlier one

Comment (girishramnani) [reply pkg/util/file_indexer.go]: This is just DRYing the code

Comment (mik-dass) [reply pkg/util/file_indexer.go]: But some comments will benefit in the future

Comment (girishramnani) [reply pkg/util/file_indexer.go]: there is nothing different or better about this approach other than that its less code then before :sweat_smile:

Comment (girishramnani) [reply pkg/util/file_indexer.go]: and at other places like this https://github.com/openshift/odo/pull/1982#discussion_r310565470. I have provided the reasoning

**Pull request merged**